// @workspace/api-client-react/index.ts
// Stub module for @workspace/api-client-react
// Exporting all necessary stub functions directly.

export function setAuthTokenGetter(getter: () => string | null): void {
  (window as any).__authTokenGetter = getter;
}

// Notification hooks
export function useGetNotifications(_: { query: any }) {
  return { data: [] as any[], refetch: () => {} };
}
export function getGetNotificationsQueryKey() { return ["notifications"]; }
export function useGetNotificationUnreadCount(_: { query: any }) {
  return { data: { count: 0 }, refetch: () => {} };
}
export function getGetNotificationUnreadCountQueryKey() { return ["notificationUnreadCount"]; }
export function useMarkNotificationRead(_: { mutation: any }) { return { mutate: () => {} }; }
export function useMarkAllNotificationsRead(_: { mutation: any }) { return { mutate: () => {} }; }

// Business hooks
export function getGetBusinessesQueryKey() { return ["businesses"]; }
export function getGetFeaturedBusinessesQueryKey() { return ["featuredBusinesses"]; }
export function getGetFeaturedTopBusinessesQueryKey() { return ["featuredTopBusinesses"]; }
export function getGetBusinessQueryKey(businessId: number) { return ["business", businessId]; }
export function useCreateBusiness(_: any) { return { mutate: () => {} }; }
export function useGetBusinesses(_: any) { return { data: [] as any[], refetch: () => {} }; }
export function useGetFeaturedBusinesses(_: any) { return { data: [] as any[], refetch: () => {} }; }
export function useGetFeaturedTopBusinesses(_: any) { return { data: [] as any[], refetch: () => {} }; }
export function useGetStatsSummary(_: any) { return { data: null, refetch: () => {} }; }
export function useGetFavorites(_: any) { return { data: [] as any[], refetch: () => {} }; }
export function getGetFavoritesQueryKey() { return ["favorites"]; }
export function useGetCategories(_: any) { return { data: [] as any[], refetch: () => {} }; }
export function useListProducts(_: any) { return { data: [] as any[], refetch: () => {} }; }
export function getListProductsQueryKey() { return ["listProducts"]; }
export function useGetDashboardStats(_: any) { return { data: null, refetch: () => {} }; }
export function useGetBusinessAnalytics(_: any) { return { data: null, refetch: () => {} }; }
export function getGetBusinessAnalyticsQueryKey() { return ["businessAnalytics"]; }
export function useGetAdminBusinesses(_: any) { return { data: [] as any[], refetch: () => {} }; }
export function useVerifyBusiness(_: any) { return { mutate: () => {} }; }
export function useCreateOrder(_: any) { return { mutate: () => {}, isPending: false }; }
export function useTrackEvent(_: any) { return { mutate: () => {} }; }
export function useTrackDelivery(_: any) { return { mutate: () => {} }; }
export function useGetProduct(_: any) { return { data: null, refetch: () => {} }; }
export function useToggleFavorite(_: any) { return { mutate: () => {} }; }
export function useVerifyPaystackPayment(_: any) { return { mutate: () => {} }; }
export function useRegister(_: any) { return { mutate: () => {} }; }
export function useLogin(_: any) { return { mutate: () => {} }; }
export function useUpdateProfile(_: any) { return { mutate: () => {} }; }
export function useChangePassword(_: any) { return { mutate: () => {} }; }
export function useDeleteAccount(_: any) { return { mutate: () => {} }; }

// Types
export type Business = any;
export type Product = any;

// User orders
export function useGetUserOrders(_: any) { return { data: [] as any[], refetch: () => {} }; }
export function getGetUserOrdersQueryKey() { return ["userOrders"]; }

// Business products hooks
export function getGetBusinessProductsQueryKey(businessId: number) { return ["businessProducts", businessId]; }
export function useGetBusinessProducts(businessId: number, options?: any) { return { data: [] as any[], refetch: () => {} }; }

// Product management hooks
export function useCreateProduct(_: any) { return { mutate: () => {}, isPending: false }; }
export function useUpdateProduct(_: any) { return { mutate: () => {}, isPending: false }; }
export function useDeleteProduct(_: any) { return { mutate: () => {} }; }
// Business reviews hooks
export function getGetBusinessReviewsQueryKey(businessId: number) { return ["businessReviews", businessId]; }
export function useGetBusinessReviews(businessId: number, options?: any) { return { data: [] as any[], refetch: () => {} }; }
